python/pyspark/ml/pipeline.py

```diff
@@ -89,9 +87,9 @@ def setParams(self, stages=None):
         setParams(self, stages=None)
         Sets params for Pipeline.
         """
-        if stages is None:
-            stages = []
         kwargs = self.setParams._input_kwargs
+        if stages is None:
+            kwargs['stages'] = []
```

Comment:
Yah I guess since it was previously being checked it might be good to check (although switching it to only do this when explicitly set to none as your editted comment does probably makes sense). Maybe @sethah and or @davies / @yanboliang can weigh in?